### PR TITLE
Add keras.ops.searchsorted

### PR DESCRIPTION
**Pull request by LarsKue**
This is commonly used for spline transformations.

### CODE REVIEW

**Comment by LarsKue** [comment]
Converted to draft because I will add a test

**Comment by LarsKue** [comment]
I opted to try to maximize support for N-D searchsorted (because this is my use-case). However, numpy does not support it. JAX supports it by vmapping, which I implemented.

If you have better suggestions on how we can support N-D searchsorted, I would be happy to implement them.

**Comment by fchollet** [comment]
Thanks for the PR!

> I opted to try to maximize support for N-D searchsorted (because this is my use-case). However, numpy does not support it. JAX supports it by vmapping, which I implemented.

Since we support vmapping APIs, we could simply not implement N-D support for this op. We should try to stay as close to NumPy as possible in order to minimize user surprise.

A TF test seems to be failing:

```
>           assert knp.all(knp.searchsorted(a, v) == expected)

keras/src/ops/numpy_test.py:3990: 
...
E     tensorflow.python.framework.errors_impl.InvalidArgumentError:
cannot compute Equal as input #1(zero-based) was expected to be a int32 tensor but is a int64 tensor
```

**Comment by LarsKue** [comment]
> we could simply not implement N-D support for this op.
You make a good point. In that case, should we raise an error if the user passes an N-D `sorted_sequence` then, or let the backend handle it if it is incompatible? Raising would make the function truly agnostic, but prevent using built-in N-D support for backends that do support it.

> A TF test seems to be failing
We can drop the part that is failing if we only support 1-D.

File: keras/src/ops/numpy_test.py [hunk 1]
```diff
@@ -3966,6 +3966,13 @@ def test_round(self):
         self.assertAllClose(knp.round(x, decimals=-1), np.round(x, decimals=-1))
         self.assertAllClose(knp.Round(decimals=-1)(x), np.round(x, decimals=-1))
 
+    def test_searchsorted(self):
```
Comment (fchollet) [inline keras/src/ops/numpy_test.py:3969]: Please also add a test for static shape inference (on KerasTensors).

**Comment by fchollet** [comment]
>  In that case, should we raise an error if the user passes an N-D sorted_sequence, or let the backend handle it if it is incompatible?

Better to do it in each backend function I think!

File: keras/src/ops/numpy_test.py (hunk 1 above)
Comment (LarsKue) [reply keras/src/ops/numpy_test.py:3975]: Done

**Comment by LarsKue** [comment]
@fchollet I implemented your requested changes, raising the error in each backend function, respectively. However, I currently do not know of a way to raise the error appropriately in the `SearchSorted.symbolic_call` method. This is for two reasons:
1. `ndim` returns different output for symbolic vs. non-symbolic tensors
2. `symbolic_call` is called with both symbolic and non-symbolic input (see e.g. the `NumpyDtypeTest` tests)

**Comment by fchollet** [comment]
Rather than using the function `ndim()` you could use `len(shape(x))`, which works the same everywhere?

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

**Pull request merged**